src/libarena/lib.rs

```diff
@@ -99,6 +99,7 @@ impl<T> TypedArenaChunk<T> {
 }
 
 const PAGE: usize = 4096;
+const BIG_CHUNK: usize = 1024 * 1024;
```

Comment:
Have you tried 2MB? That would match the huge page size on x86 and benefit from THP optimization, at least on `transparent_hugepage/enabled = always` systems.

Reply:
I haven't, but I'm happy to change it to 2 MiB. I'll add a comment explaining why.